### PR TITLE
More unit tests for VisitorList, VisitorMap, and VisitorSet

### PR DESCRIPTION
**Pull request by ghost**
Unit tests to contribute to #820. 

The `toString` test for VisitorSet currently fails - this was intentional. From the looks of things, the current implementation deletes the last character for the last element unintentionally. I have just written the test based on how `toString` works for the other classes - that is, there should be a space after the commas separating the elements. I figured having that space would fix the problem of the missing character since that space would get deleted instead. 

If you would like this test changed however (or any for that matter) I'd be happy to.

### CODE REVIEW

File: javaparser-core-testing/src/test/java/com/github/javaparser/utils/VisitorSetTest.java [hunk 1]
```diff
@@ -138,4 +137,42 @@ void visitorSetToArray() {
...
+                new ObjectIdentityEqualsVisitor());
+        assertTrue(set.isEmpty());
+
+        set.addAll(list);
+        list.add(parse("class X{}"));
+        assertFalse(set.containsAll(list));
+    }
+
+    @Test
+    void visitorSetToString() {
```
Comment (jlerbsc) [inline javaparser-core-testing/src/test/java/com/github/javaparser/utils/VisitorSetTest.java:159]: Maybe you can use Parameterized  tests here

File: javaparser-core-testing/src/test/java/com/github/javaparser/utils/VisitorSetTest.java [hunk 2]
```diff
@@ -138,4 +137,42 @@ void visitorSetToArray() {
         for (CompilationUnit u : set.toArray(new CompilationUnit[2]))
             assertTrue(set.contains(u));
     }
+
+    @Test
+    void containsAllFalse() {
```
Comment (jlerbsc) [inline javaparser-core-testing/src/test/java/com/github/javaparser/utils/VisitorSetTest.java]: Please can you keep the method naming convention used in this test class?
Can you rename this method for example visitorSetContainsOnlyPart.

File: javaparser-core-testing/src/test/java/com/github/javaparser/utils/VisitorSetTest.java [hunk 3]
```diff
@@ -162,7 +162,7 @@ void visitorSetToString() {
         CompilationUnit x2 = parse("class Y{}");
         set.add(x2);
 
-        String[] validStrings = new String[]{"[" + x1 + "," + x2 + "]", "[" + x2 + "," + x1 + "]"};
+        String[] validStrings = new String[]{"[" + x1 + ", " + x2 + "]", "[" + x2 + ", " + x1 + "]"};
```
Comment (jlerbsc) [inline javaparser-core-testing/src/test/java/com/github/javaparser/utils/VisitorSetTest.java]: I think it is this pattern that should be modified by using for example the join functionality in streams.

> for (EqualsHashcodeOverridingFacade facade : innerList) {
            sb.append(facade.overridden.toString() + ", ");
        }
        return sb.replace(sb.length() - 2, sb.length(), "]").toString();

Comment (ghost) [reply javaparser-core-testing/src/test/java/com/github/javaparser/utils/VisitorSetTest.java]: To clarify sir, when you say the pattern that should be modified, are you referring to the `StringBuilder` for loop in the `toString` method or the way `validStrings` are created in the test? Or perhaps both should be modified?

Comment (jlerbsc) [reply javaparser-core-testing/src/test/java/com/github/javaparser/utils/VisitorSetTest.java]: Hi @flanbino both should be modified in particular i refer to this code that add a separator and finally replace the last separator.

Comment (ghost) [reply javaparser-core-testing/src/test/java/com/github/javaparser/utils/VisitorSetTest.java]: Hello again, I've made the changes requested

**Comment by jlerbsc** [comment]
Sounds good to me.

**Pull request closed without merging**